### PR TITLE
csv: delimiter supports at most 2 characters

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/9969

### What is changed and how it works?

CSV delimiter supports at most 2 characters and to support hex delimiters like "FE" or "FF"

#### Tests 

 - Unit test

##### Will it cause performance regression or break compatibility?

No

##### Do you need to update user documentation, design documentation or monitoring documentation?

Yes, TODO

### Release note 

```release-note
Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

If you don't think this PR needs a release note then fill it with `None`.
```
TiCDC sink to CSV customized delimiter supports at most 2 characters.

### CODE REVIEW

**Review by CharlesCheung96 — APPROVED** [review]

File: pkg/config/sink_test.go [hunk 1]
```diff
@@ -350,20 +359,28 @@ func TestValidateAndAdjustCSVConfig(t *testing.T) {
...
 				Quote:     "'",
 				Delimiter: "'",
 			},
-			wantErr: "csv config quote and delimiter cannot be the same",
+			wantErr: "csv config quote and delimiter has common characters which is not allowed",
+		},
+		{
+			name: "delimiter and quote contain common characters",
+			config: &CSVConfig{
+				Quote:     "EX",
```
Comment (CharlesCheung96) [inline pkg/config/sink_test.go:380]: Quote is limited to one character.

Comment (zhangjinpeng1987) [reply pkg/config/sink_test.go]: nice catch, fixed

**Review by nongfushanquan — APPROVED** [review]

**Pull request merged**